### ISSUE TITLE
push_patch does not support routes with scope host

### ISSUE THREAD

**Issue by mrdotb** [phoenixframework/phoenix_live_view#1836]
Hello,

It's the same as #885 this issue appear to be fixed but I am not 100% sure.

### Environment

```
Erlang/OTP 24 [erts-12.1.5] [source] [64-bit] [smp:24:24] [ds:24:24:10] [async-threads:1] [jit]
Elixir 1.12.3 (compiled with Erlang/OTP 24)
```
* Phoenix version (mix deps):
* Phoenix LiveView version (mix deps):
```
{:phoenix, "~> 1.6.6"},
{:phoenix_live_view, "~> 0.17.5"},
```
* Operating system:
ubuntu 20.04

### Actual behavior

```elixir
  scope "/", ArkadyWeb, host: "arkady.local" do
    pipe_through :browser

    live "/", Live.Book.Index, :index
    live "/:id", Live.Book.Index, :show
  end
```
```elixir
    to = Routes.book_index_path(socket, :show, 123)
    push_patch(socket, to: to, replace: true)
```

```
[error] #PID<0.511.0> running ArkadyWeb.Endpoint (connection #PID<0.510.0>, stream id 1) terminated
Server: arkady.local:4000 (http)
Request: GET /
** (exit) an exception was raised:
    ** (ArgumentError) cannot invoke handle_params nor live_redirect/live_patch to "/123" because it isn't defined in ArkadyWeb.Router
        (phoenix_live_view 0.17.5) lib/phoenix_live_view/route.ex:50: Phoenix.LiveView.Route.live_link_info!/3
        (phoenix_live_view 0.17.5) lib/phoenix_live_view.ex:1052: Phoenix.LiveView.push_patch/2
        (arkady 0.1.0) lib/arkady_web/live/book/index.ex:9: ArkadyWeb.Live.Book.Index.handle_params/3
        (phoenix_live_view 0.17.5) lib/phoenix_live_view/utils.ex:369: anonymous fn/5 in Phoenix.LiveView.Utils.call_handle_params!/5
        (telemetry 1.0.0) /home/john/Projects/elixir/arkady/deps/telemetry/src/telemetry.erl:293: :telemetry.span/3
        (phoenix_live_view 0.17.5) lib/phoenix_live_view/static.ex:261: Phoenix.LiveView.Static.call_mount_and_handle_params!/5
        (phoenix_live_view 0.17.5) lib/phoenix_live_view/static.ex:110: Phoenix.LiveView.Static.render/3
        (phoenix_live_view 0.17.5) lib/phoenix_live_view/controller.ex:38: Phoenix.LiveView.Controller.live_render/3
        (phoenix 1.6.6) lib/phoenix/router.ex:355: Phoenix.Router.__call__/2
        (arkady 0.1.0) lib/arkady_web/endpoint.ex:1: ArkadyWeb.Endpoint.plug_builder_call/2
        (arkady 0.1.0) lib/plug/debugger.ex:136: ArkadyWeb.Endpoint."call (overridable 3)"/2
        (arkady 0.1.0) lib/arkady_web/endpoint.ex:1: ArkadyWeb.Endpoint.call/2
        (phoenix 1.6.6) lib/phoenix/endpoint/cowboy2_handler.ex:54: Phoenix.Endpoint.Cowboy2Handler.init/4
        (cowboy 2.9.0) /home/john/Projects/elixir/arkady/deps/cowboy/src/cowboy_handler.erl:37: :cowboy_handler.execute/2
        (cowboy 2.9.0) /home/john/Projects/elixir/arkady/deps/cowboy/src/cowboy_stream_h.erl:306: :cowboy_stream_h.execute/3
        (cowboy 2.9.0) /home/john/Projects/elixir/arkady/deps/cowboy/src/cowboy_stream_h.erl:295: :cowboy_stream_h.request_process/3
        (stdlib 3.16.1) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
```

### Expected behavior
`push_patch` in the same host should work.

I made a repo to show the bug.
https://github.com/mrdotb/arkady

**Comment by josevalim** [comment]
Can you please try out master too? There were some recent bugs related to this, thanks!

**Comment by mrdotb** [comment]
It's working with liveview from master so I guess it should be fixed in the next release.

**Comment by mrdotb** [comment]
Fixed in ` {:phoenix_live_view, "~> 0.17.6"}`
